beacon-chain/execution/engine_client.go

```diff
@@ -792,23 +658,25 @@ func fullPayloadFromPayloadBody(
 		}
 		return blocks.WrappedExecutionPayloadElectra(
 			&pb.ExecutionPayloadElectra{
-				ParentHash:    header.ParentHash(),
-				FeeRecipient:  header.FeeRecipient(),
-				StateRoot:     header.StateRoot(),
-				ReceiptsRoot:  header.ReceiptsRoot(),
-				LogsBloom:     header.LogsBloom(),
-				PrevRandao:    header.PrevRandao(),
-				BlockNumber:   header.BlockNumber(),
-				GasLimit:      header.GasLimit(),
-				GasUsed:       header.GasUsed(),
-				Timestamp:     header.Timestamp(),
-				ExtraData:     header.ExtraData(),
-				BaseFeePerGas: header.BaseFeePerGas(),
-				BlockHash:     header.BlockHash(),
-				Transactions:  body.Transactions,
-				Withdrawals:   body.Withdrawals,
-				ExcessBlobGas: ebg,
-				BlobGasUsed:   bgu,
+				ParentHash:         header.ParentHash(),
+				FeeRecipient:       header.FeeRecipient(),
+				StateRoot:          header.StateRoot(),
+				ReceiptsRoot:       header.ReceiptsRoot(),
+				LogsBloom:          header.LogsBloom(),
+				PrevRandao:         header.PrevRandao(),
+				BlockNumber:        header.BlockNumber(),
+				GasLimit:           header.GasLimit(),
+				GasUsed:            header.GasUsed(),
+				Timestamp:          header.Timestamp(),
+				ExtraData:          header.ExtraData(),
+				BaseFeePerGas:      header.BaseFeePerGas(),
+				BlockHash:          header.BlockHash(),
+				Transactions:       pb.RecastHexutilByteSlice(body.Transactions),
+				Withdrawals:        body.Withdrawals,
+				ExcessBlobGas:      ebg,
+				BlobGasUsed:        bgu,
+				DepositReceipts:    pb.JsonDepositRequestsToProto(body.DepositRequests),
+				WithdrawalRequests: pb.JsonWithdrawalRequestsToProto(body.WithdrawalRequests),
 			}, big.NewInt(0)) // We can't get the block value and don't care about the block value for this instance
```

Comment:
I know this isn't code you changed, but why can't we get the value? `header.ValueInWei()` or the gwei version?

Reply:
It doesn't exist because it isn't part of the payload, it is a separate field in the new payload response or builder api response for a blinded block bid. This doesn't really belong in the ExecutionData interface. It's a hack for bundling values together during proposals. I actually have another branch to remove this value from the `ExecutionData`.